src/driver_client/dclient.rs

```diff
@@ -206,12 +218,14 @@ impl DriverClient {
             HBICAP_ADDR::ADDR_CPU2HIF_HBICAP_TRANSFER_SIZE,
             (binary.len() / 4) as u32,
         )?;
-        self.dma_write_by_chunks(self.cfg.ctrl_hbicap_baseaddr, DMA_RW::OFFSET, binary, 4)?;
-
+        self.dma_write(self.cfg.dma_hbicap_baseaddr, DMA_RW::OFFSET, binary)?;
         while !self.is_hbicap_ready() {
-            sleep(Duration::from_millis(1));
+            sleep(Duration::from_millis(10));
```

Comment:
why do we need to wait if HBI isn't ready?

Reply:
we're waiting loading binary here

Reply:
is there a way to detect this without sleep?

Reply:
but yeah if there is no way to avoid sleep, I suggest a detailed comment to avoid confusion later on.